app/components/list-widget/component.js

```diff
@@ -66,6 +70,15 @@ export default Ember.Component.extend({
             contentType: 'application/json'
         });
         this.set("data", JSON.parse(agent_details));
+
+        let d = []; 
+        this.get('data').forEach(function(element) {
+            if(!(element.name === "")){
+                d.push(element);
+            }
+        });
+        this.set("data", d);
```

Comment:
This can be made a lot more succinct using a filter

this.set('data', JSON.parse(agent_details).filter(element => element.name !== ''))